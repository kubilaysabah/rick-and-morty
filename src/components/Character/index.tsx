// Next
import { useRouter } from "next/router";

// React
import type { FC, ReactElement } from "react";

// Style
import Style from "./style.module.css";

const Character: FC<ICharacter> = ({
    created,
    episode,
    gender,
    id,
    image,
    location,
    name,
    origin,
    species,
    status,
    type,
    url,
}: ICharacter): ReactElement<ICharacter> => {
    const router = useRouter();

    return (
        <div className={Style.content}>
            <div>
                <div className={Style.contentContainer}>
                    <div className={Style.imageColumn}>
                        <img src={image} alt={name} className={Style.image} />
                    </div>
                    <div className={Style.contentColumn}>
                        <div>
                            <h1 className={Style.title}>{name}</h1>
                            <div className={Style.statusContainer}>
                                <div className={`${status === "Alive" ? Style.statusAlive : status === "Dead" ? Style.statusDead : Style.statusUnknown}`} />
                                <div className={Style.statusText}>{status}</div>
                            </div>
                            <small>{location.name} - {type} - {species} - {gender}</small>
                        </div>
                        <button className={Style.turnBackButton} onClick={() => router.back()}>Geri DÃ¶n</button>
                    </div>
                </div>
            </div>
        </div>
    );
};


export default Character;