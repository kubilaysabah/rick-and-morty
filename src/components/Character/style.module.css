.content {
    @apply h-screen flex items-center justify-center;
}

.contentContainer {
    @apply flex flex-wrap;
}

.contentColumn {
    @apply flex-1 ml-3 text-white;
}

.image {
    @apply max-w-full;
}

.title {
    @apply font-bold text-2xl;
}


.statusContainer {
    @apply flex items-center;
}

.status {
    @apply w-2 h-2 rounded-full;
}

.statusAlive {
    @apply status bg-green-600;
}

.statusDead {
    @apply status bg-red-600;
}

.statusUnknown {
    @apply status bg-yellow-600;
}

.statusText {
    @apply text-white ml-2;
}

.turnBackButton {
    @apply focus:outline-none bg-gray-300 text-gray-700 py-2 text-center rounded block font-semibold mt-3;
}