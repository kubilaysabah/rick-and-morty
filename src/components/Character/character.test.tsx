import renderer from "react-test-renderer";
import Character from ".";

const Props = {
    status: "Alive" as "Alive" | "Dead" | "unknown",
    created: "2017-11-04T18:48:46.250Z",
    episode: ["https://rickandmortyapi.com/api/episode/1"],
    gender: "Male" as "Male" | "Female" | "Genderless",
    id: 1,
    image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
    location: {
        name: "Citadel of Ricks",
        url: "https://rickandmortyapi.com/api/location/3"
    },
    name: "Rick Sanchez",
    origin: {
        name: "Citadel of Ricks",
        url: "https://rickandmortyapi.com/api/location/3"
    },
    species: "Human",
    type: "Human",
    url: "https://rickandmortyapi.com/api/character/1"
}

it('renders correctly', () => {
    const tree = renderer
        .create(<Character {...Props} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});