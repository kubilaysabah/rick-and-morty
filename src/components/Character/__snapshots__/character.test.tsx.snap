// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="content"
>
  <div>
    <div
      className="contentContainer"
    >
      <div
        className="imageColumn"
      >
        <img
          alt="Rick Sanchez"
          className="image"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
      </div>
      <div
        className="contentColumn"
      >
        <div>
          <h1
            className="title"
          >
            Rick Sanchez
          </h1>
          <div
            className="statusContainer"
          >
            <div
              className="statusAlive"
            />
            <div
              className="statusText"
            >
              Alive
            </div>
          </div>
          <small>
            Citadel of Ricks
             - 
            Human
             - 
            Human
             - 
            Male
          </small>
        </div>
        <button
          className="turnBackButton"
          onClick={[Function]}
        >
          Geri DÃ¶n
        </button>
      </div>
    </div>
  </div>
</div>
`;
